src/components/FilteredSearchBar.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+import React, { PureComponent, Fragment } from "react";
+import { StyleSheet, TextInput, View, TouchableOpacity } from "react-native";
+import throttle from "lodash/throttle";
+
+import Search from "./Search";
+import SearchIcon from "../images/icons/Search";
+import Close from "../images/icons/Close";
+
+import colors from "../colors";
+
+type Props = {
+  renderList: (list: Array<*>) => React$Node,
+  renderEmptySearch: () => React$Node,
+  keys?: Array<string>,
+  list: Array<*>,
+};
+
+type State = {
+  focused: boolean,
+  query: string,
+};
+
+class FilteredSearchBar extends PureComponent<Props, State> {
+  static defaultProps = {
+    keys: ["name"],
+  };
+
+  state = {
+    focused: false,
+    query: "",
+  };
+
+  constructor(props: *) {
+    super(props);
+
+    this.onChange = throttle(this.onChange, 200);
```

Comment:
you could inline `onChange = throttle((text:string) => { this.setState(...) }, 200)` down there so you don't need constructor ;)

also, it would be a good practice to have a componentWillMount that will call `this.onChange.cancel()`

Reply:
fixing this right now